engines/bops_api/app/controllers/concerns/bops_api/error_handler.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module BopsApi
+  module ErrorHandler
+    extend ActiveSupport::Concern
+
+    EXCEPTIONS = {
+      "ActiveRecord::RecordInvalid" => {status: 400, message: "Record Invalid"},
+      "InvalidSchemaError" => {status: 400, message: "Bad Request"},
+      "NotPermittedError" => {status: 403, message: "Forbidden"},
+      "ActiveRecord::RecordNotFound" => {status: 404, message: "Cannot find record"},
+      "InternalServerError" => {status: 500, message: "Internal Server Error"}
+    }.freeze
+
+    class BaseError < StandardError; end
+
+    EXCEPTIONS.each_key do |exception_name|
+      const_set(exception_name, Class.new(BaseError)) unless const_defined?(exception_name)
+    end
```

Comment:
This looks like you're defining these errors inside `ErrorHandler`? Probably better to define them inside an Errors constant and statically in a file in app/lib/bops_api/errors.rb, e.g:

``` ruby
module BopsApi
  module Errors
    class BaseError < ::StandardError; end
    class InvalidSchemaError < BaseError; end
    class NotPermittedError < BaseError; end
    class InternalServerError < BaseError; end
  end
end
```